test/integration/integration_test.cc

```diff
@@ -237,7 +237,9 @@ TEST_P(IntegrationTest, InvalidContentLength) { testInvalidContentLength(); }
 
 TEST_P(IntegrationTest, MultipleContentLengths) { testMultipleContentLengths(); }
 
-TEST_P(IntegrationTest, OverlyLongHeaders) { testOverlyLongHeaders(); }
+TEST_P(IntegrationTest, LargeHeadersRejected) { testLargeRequestHeaders(60, 60); }
+
+TEST_P(IntegrationTest, LargeHeadersAccepted) { testLargeRequestHeaders(60, 62); }
```

Comment:
can you make this 63, and add this one to http2_integration_test as well?  before we were only pushing the hcm limits so having a single test was fine, but it'd be good to have a regression test for making sure the max size passes through both H1 and H2 codecs.  

If you want bonus points you could the max value, and use that here and in a config rejected test, but I think the detailed comments you have in the api should suffice :-)

Reply:
I can't make it 63, because the actual byte size would be 63K + misc and exceed the max configurable. I made the outgoing request header size consistent across tests though, and added h2 tests. (Had to write out the rejected case because it expected a stream reset rather than a disconnect.